[16.0][IMP] account_payment_purchase: Propagate PO payment mode when invoicing

aritzolea opened the pull request:
Comes from https://github.com/OCA/bank-payment/pull/1225

Fixes payment mode propagation from purchase order to invoice when supplier's payment mode is different to the one on the purchase order.

aritzolea committed:
[FIX] account_payment_purchase: Add propagation of payment mode from purchase to invoice

Fixes payment mode propagation from purchase order to invoice when supplier's payment mode is different to the one on the purchase order.

Co-authored-by: atorreiro <atorreiro@kernet.es>

aritzolea commented:
Could you check this @pedrobaeza? Thanks in advance

pedrobaeza commented on account_payment_purchase/models/purchase_order.py:
```diff
@@ -56,7 +56,10 @@ def _compute_payment_mode(self):
         invoice_vals = super()._prepare_invoice()
         invoice_vals.pop("partner_bank_id")
+        avoid_assign_pay_mode = self._context.get("avoid_pay_mode_propagate", False)
+        if not avoid_assign_pay_mode:
```
Instead of having `not` and a "negative" variable, better to invert it (and the same for the rest):

```suggestion
        if assign_payment_mode:
```

aritzolea replied:
Reverted the variable, now it is possitive.

pedrobaeza reviewed (changes requested):
And why it was not propagated the payment mode? Please document properly the commit message and the PR description.

aritzolea commented:
> And why it was not propagated the payment mode? Please document properly the commit message and the PR description.

I don't know if it was a bug or there was a specific reason for not propagating the payment mode, I could edit the commit message explaning mode is now propagated, but I don't know if that's what you mean.

aritzolea commented:
> > And why it was not propagated the payment mode? Please document properly the commit message and the PR description.
> 
> I don't know if it was a bug or there was a specific reason for not propagating the payment mode, I could edit the commit message explaning mode is now propagated, but I don't know if that's what you mean.

I added this commit message: 

> When invoicing a purchase order the payment mode was not being propagated prior
to this commit.

Do you think it's now OK for merge or would you add something else on commit message @pedrobaeza ?

pedrobaeza commented:
3 commits is too much for such little code and being about the same problem, but I still don't find in your PR description nor the commit messages the steps to reproduce the problem.

aritzolea commented:
> 3 commits is too much for such little code and being about the same problem, but I still don't find in your PR description nor the commit messages the steps to reproduce the problem.

Squashed commits and added co-author.

Explained on PR description and commit message the case in which this bug is fixed.

pedrobaeza commented:
I have checked the code, and I still don't get the problem, or at least, it shouldn't be fixed there. We have this code for that:

https://github.com/OCA/bank-payment/pull/1410/files#diff-9e60db0f5d8d1708de9fd29c9e6c32f0622a159c541173ab6c3d51ce1d8e3bcbR18-R33

Please specify complete steps to reproduce the problem.

aritzolea commented:
> I have checked the code, and I still don't get the problem, or at least, it shouldn't be fixed there. We have this code for that:
> 
> https://github.com/OCA/bank-payment/pull/1410/files#diff-9e60db0f5d8d1708de9fd29c9e6c32f0622a159c541173ab6c3d51ce1d8e3bcbR18-R33
> 
> Please specify complete steps to reproduce the problem.

New test code would fail without the changes in the PR
